packages/wallets/src/features/cashier/modules/TransactionStatus/utils/transactionDisplayMapper.ts

```diff
@@ -0,0 +1,130 @@
+import moment from 'moment';
+import { getTruncatedString } from '@deriv/utils';
+import { THooks } from '../../../../../types';
+
+const transactionDetailsMapper = (transaction: THooks.CryptoTransactions) => {
+    const transactionHashObscure = transaction.transaction_hash
+        ? `${getTruncatedString(transaction.transaction_hash, { length: 8, type: 'middle' })}`
+        : 'Pending';
+
+    if (transaction.transaction_type === 'deposit') {
+        return {
+            CONFIRMED: {
+                confirmationDisplay: 'Confirmed',
+                description: 'Your deposit is successful.',
+                statusColor: 'successful',
+                statusName: 'Successful',
+                transactionHashDisplay: transactionHashObscure,
+            },
+            ERROR: {
+                confirmationDisplay: 'NA',
+                description:
+                    'Your deposit is unsuccessful due to an error on the blockchain. Please contact your crypto wallet service provider for more info.',
+                statusColor: 'unsuccessful',
+                statusName: 'Unsuccessful',
+                transactionHashDisplay: 'NA',
+            },
+            PENDING: {
+                confirmationDisplay: transaction.confirmations ? `${transaction.confirmations}` : 'Pending',
+                description: 'We’ve received your request and are waiting for more blockchain confirmations.',
+                statusColor: 'warning',
+                statusName: 'In process',
+                transactionHashDisplay: transactionHashObscure,
```

Comment:
that sounds like a component responsiblity to decide if something should be displayed,

Reply:
I agree, there's just too many variations and to counter this we would probably need creating a bunch of new smaller components 🤔